### PR TITLE
Add api constraints and groups controller

### PR DESCRIPTION
**Pull request by thefactus**

### CODE REVIEW

File: app/controllers/api/v1/groups_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class Api::V1::GroupsController < ApplicationController
+  def index
+    @groups = Group.all
+
+    render json: @groups
+  end
+
+  def show
+    @group = Group.find(params[:id])
```
Comment (marceloboth) [inline app/controllers/api/v1/groups_controller.rb]: `@group = Group.find(params[:id])` is repeating in action `update` and `destroy`. For avoiding this repetition, you can create a `before_action` with a private method and put this code inside. Is a tip that a usually use in my projects 😄 .

**Comment by marceloboth** [comment]
Nice job man 👍

File: app/controllers/api/v1/groups_controller.rb (hunk 1 above)
Comment (thefactus) [inline app/controllers/api/v1/groups_controller.rb]: Hi @marceloboth,
I used to do it,
but sometimes people get confused wondering where is @groups from

So nowadays I do this way:
http://craftingruby.com/posts/2015/05/31/dont-use-before-action-to-load-data.html
http://blog.ragnarson.com/2015/09/30/problems-with-typical-rails-controllers-and-before-actions.html

What do you think?

Thanks for your feedback.

Comment (marceloboth) [inline app/controllers/api/v1/groups_controller.rb]: Yeah, after i comment here, i searched about and i read this two blog posts, and now i agree about the way that you did. It's better the code seems obvious, and with this before action in some cases can be confused.

**Pull request merged**